core/ir/instr_api.h

```diff
@@ -289,6 +289,8 @@ struct _instr_t {
 
     uint opcode;
 
+    uint category;
```

Comment:
The instr_t size is tightly controlled and tuned for in the heap allocation buckets.  Adding a field requires making sure the BLOCK_SIZES in core/heap are sane, updating the commented sizes, and as a binary compatibility breakage this requires a major version bump and a listing in the release notes.

One alternative is not store anything and encode on demand when the category is asked about, like in instr_zeroes_ymmh() and others where we get the encoding info.  That has some downsides though of not having a target PC for PC-relative encoding differences and possible overhead: xref https://github.com/dvyukov/dynamorio/pull/1.

Another alternative is to fit the bits in an existing field like prefixes which is where the predicate bits are: but while they may well fit on AArch64 I think x86 uses a lot of those bits for internal decoding even if none are exposed.